src/compiler/diagnosticMessages.json

```diff
@@ -2477,5 +2477,10 @@
     "A type assertion expression is not allowed in the left-hand side of an exponentiation expression. Consider enclosing the expression in parentheses.": {
       "category": "Error",
       "code": 17007
+    },
+    "Potentially wrong empty statement in THEN clause.": {

```

Comment:
Thanks. However, Daniel expressed some concerns (https://github.com/Microsoft/TypeScript/pull/5375#issuecomment-150637784) about merging this pull request. So I don't know if I should continue in the work or not.